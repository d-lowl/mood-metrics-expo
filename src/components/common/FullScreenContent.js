import React, { Component } from 'react';
import { View, Dimensions } from 'react-native';
import { Content } from 'native-base';


class FullScreenContent extends Component {
  render() {
    const {height: screenHeight, width: screenWidth} = Dimensions.get('window');
    var style = {
      flex: 1,
      height: screenHeight - 64,
      justifyContent: 'center',
      alignItems: 'center'
    };
    if (this.props.fullWidth) {
      style = {
        ...style,
        width: screenWidth
      }
    }
    if (this.props.highlight) {
       style = {
         ...style,
         backgroundColor: 'lightgreen'
       }
    }

    //TODO Screen height fix for android and iOS
    return (
      <Content scrollEnabled={false}>
        <View style={style}>
          {this.props.children}
        </View>
      </Content>
    );
  }
}

export default FullScreenContent;
