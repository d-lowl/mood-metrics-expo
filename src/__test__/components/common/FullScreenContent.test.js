import 'react-native';
import React from 'react';
import FullScreenContent from '../../../components/common/FullScreenContent';
import { Text } from 'native-base';

import renderer from 'react-test-renderer';

test('FullScreenContent renders correctly', () => {
  const tree = renderer.create(
    <FullScreenContent />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('FullScreenContent with full width renders correctly', () => {
  const tree = renderer.create(
    <FullScreenContent fullWidth />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('FullScreenContent highlighed renders correctly', () => {
  const tree = renderer.create(
    <FullScreenContent highlight />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
