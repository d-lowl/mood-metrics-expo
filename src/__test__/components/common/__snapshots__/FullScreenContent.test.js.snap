// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenContent highlighed renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "lightgreen",
          "flex": 1,
          "height": 1270,
          "justifyContent": "center",
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`FullScreenContent renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 1270,
          "justifyContent": "center",
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`FullScreenContent with full width renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 1270,
          "justifyContent": "center",
          "width": 750,
        }
      }
    />
  </View>
</RCTScrollView>
`;
